tests/execution_engine/test_execution_engine.py

```diff
@@ -187,60 +187,66 @@ def test_resolve_metrics_with_aggregates_and_column_map():
         metric_name="column.mean",
         metric_domain_kwargs={"column": "a"},
         metric_value_kwargs=None,
-        metric_dependencies={
-            "table.columns": table_columns_metric,
-        },
     )
+    mean.metric_dependencies = {
```

Comment:
If we are constantly having to initialize `metric_dependencies` after the call to `init`, then it really should be part of `init`.

Reply:
@donaldheppner No, because they are NEVER known at time of `init` -- in all uses of MetricConfiguration as a constructor, the metric_dependencies is either omitted or set to None — having it there from the original implementation a couple of years ago, and it is ALWAYS None (because there is no knowledge of what the dependencies should be at the init/constructor time).

Reply:
>No, because they are NEVER known at time of init

@alexsherstinsky but is that just incidental to how we happen to be using it currently? Or will that always be true?

Reply:
@Kilo59 It has always been true and will continue being so on the account of where/how dependencies get computed and become known (not at `__init__()` stage).  Thanks.